File: functions/alsa-lib/variants/go/4_val_seg_8.go
package main

func valSeg(val int32) int {
	r := 1
	val >>= 8
	bitLength := [5]int{0, 0, 1, 2, 4}
	for i := 4; i > 0; i-- {
		if val >= (1 << (bitLength[i] - 1)) {
			val >>= bitLength[i]
			r += bitLength[i]
		}
	}
	return r
}


func main(){
    valSeg(0)
}

File: functions/alsa-lib/variants/c/4_val_seg_8.c
#include <stdint.h>

static inline int val_seg(int val)
{
	int r = 1;
	val >>= 8;
	switch (val & 0xf0) {
		case 0xf0:
			val >>= 4;
			r += 4;
			break;
		default:
			break;
	}
	switch (val & 0x0c) {
		case 0x0c:
			val >>= 2;
			r += 2;
			break;
		default:
			break;
	}
	if (val & 0x02) {
		r += 1;
	}
	return r;
}


int main(){
    val_seg(0);
}
